flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/AggregatedTaskDetailsInfo.java

```diff
@@ -0,0 +1,265 @@
+                    statusDuration
+                            .get(ExecutionState.DEPLOYING.name())
+                            .addValue(status.get(ExecutionState.DEPLOYING));
+                    statusDuration
+                            .get(ExecutionState.RUNNING.name())
+                            .addValue(status.get(ExecutionState.RUNNING));
+                });
+        return new AggregatedTaskDetailsInfo(
+                metrics.values().stream()
+                        .collect(
+                                Collectors.toMap(
+                                        MetricsStatistics::getName, MetricsStatistics::toMap)),
+                statusDuration.values().stream()
+                        .collect(
+                                Collectors.toMap(
+                                        MetricsStatistics::getName, MetricsStatistics::toMap)));
+    }
+
+    @VisibleForTesting
+    static class MetricsStatistics {
+        private final List<Long> values = new ArrayList<>();
+        private final String name;
+        private boolean sorted = false;
+        private long sum = 0;
+
+        MetricsStatistics(String name) {
+            this.name = name;
+        }
+
+        void addValue(long value) {
+            values.add(value);
+            sum += value;
+            sorted = false;
+        }
+
+        private String getName() {
+            return name;
+        }
+
+        private Map<String, Long> toMap() {
+            Map<String, Long> result = new HashMap<>();
+            result.put("min", getMin());
+            result.put("max", getMax());
+            result.put("avg", getAvg());
+            result.put("sum", getSum());
+            result.put("median", getMedian());
+            result.put("p25", getP25());
+            result.put("p75", getP75());
+            result.put("p95", getP95());
+            return result;
+        }
+
+        long getMin() {
+            if (!sorted) {
+                values.sort(Comparator.naturalOrder());
+                sorted = true;
+            }
+            return values.isEmpty() ? 0 : values.get(0);
+        }
+
+        long getMax() {
+            if (!sorted) {
+                values.sort(Comparator.naturalOrder());
+                sorted = true;
+            }
+            return values.isEmpty() ? 0 : values.get(values.size() - 1);
+        }
+
+        long getMedian() {
+            if (!sorted) {
+                values.sort(Comparator.naturalOrder());
+                sorted = true;
+            }
+            return values.isEmpty() ? 0 : values.get(values.size() / 2);
+        }
+
+        long getP25() {
+            if (!sorted) {
+                values.sort(Comparator.naturalOrder());
+                sorted = true;
+            }
+            return values.isEmpty()
+                    ? 0
+                    : values.get(
+                            Math.min(values.size() - 1, (int) Math.rint(values.size() * 0.25)));
```

Comment:
This is a very common calculation. I wonder if we can leverage some existing libraries. E.g., I find `org.apache.commons.math3.stat.descriptive.rank.Percentile` which might fit our needs.